src/software/sensor_fusion/refbox_data.h

```diff
@@ -72,8 +72,90 @@ enum class RefboxStage
     REFBOX_STAGE_COUNT
 };
 
-struct TeamInfo
+class TeamInfo
 {
+   public:
+    TeamInfo() {}
+    TeamInfo(std::string name, int score, int red_cards,
+             std::vector<int> yellow_card_times, int yellow_cards, int timeouts,
+             int timeout_time, int goalkeeper, int foul_counter,
+             int ball_placement_failures, bool can_place_ball, int max_allowed_bots)
+        : name(name),
+          score(score),
+          red_cards(red_cards),
+          yellow_card_times(yellow_card_times),
+          yellow_cards(yellow_cards),
+          timeouts(timeouts),
+          timeout_time(timeout_time),
+          goalkeeper(goalkeeper),
+          foul_counter(foul_counter),
+          ball_placement_failures(ball_placement_failures),
+          can_place_ball(can_place_ball),
+          max_allowed_bots(max_allowed_bots)
+    {
+    }
+
+    std::string getName(void)
+    {
+        return name;
+    }
+
+    int getScore(void)
+    {
+        return score;
+    }
+
+    int getRedCards(void)
+    {
+        return red_cards;
+    }
+
+    std::vector<int> getYellowCardTimes(void)
+    {
+        return yellow_card_times;
+    }
+
+    int getYellowCards(void)
+    {
+        return yellow_cards;
+    }
+
+    int getTimeouts(void)
+    {
+        return timeouts;
+    }
+
+    int getTimeoutTime(void)
+    {
+        return timeout_time;
+    }
+
+    int getGoalkeeper(void)
+    {
+        return goalkeeper;
+    }
+
+    int getFoulCounter(void)
+    {
+        return foul_counter;
+    }
+
+    int getBallPlacementFailures(void)
+    {
+        return ball_placement_failures;
+    }
+
+    bool getCanPlaceBall(void)
+    {
+        return can_place_ball;
+    }
+
+    int getMaxAllowedBots(void)
+    {
+        return max_allowed_bots;
+    }
```

Comment:
can we put the implementations of these functions in a `refbox_data.cpp` file? Not that they really need to be hidden, but it's more consistent with the rest of the codebase and less distracting to read. 

Also these all need jdocs; I know they're really not needed here, but it's easier for us to have a hard rule that all functions have jdocs then to allow everyone (with a variety of experience levels) to put them in at their own discretion.

Reply:
fixed